modules/001-flow-schema/hooks/discover_d8_ns.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2023 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/go_lib/set"
+)
+
+const namespacesValuesPath = `flowSchema.internal.namespaces`
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue:        "/modules/flow-schema",
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 10},
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "namespaces",
+			ApiVersion: "v1",
+			Kind:       "Namespace",
+			LabelSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "heritage",
+						Operator: metav1.LabelSelectorOpIn,
+						Values: []string{
+							"deckhouse",
+						},
+					},
+				},
+			},
+			FilterFunc: applyNamespaceFilter,
+		},
+	},
+}, handleNamespaces)
+
+func applyNamespaceFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	return obj.GetName(), nil
+}
+
+func handleNamespaces(input *go_hook.HookInput) error {
+	snap := input.Snapshots["namespaces"]
+	if len(snap) == 0 {
+		return errors.New("deckhouse namespaces not found")
+	}
+
+	nsSlice := set.NewFromSnapshot(snap).Slice()
+	sort.Slice(nsSlice, func(i, j int) bool {
```

Comment:
1. `sort.Strings` is better for this case.
2. nsSlice is already sorted. See https://github.com/deckhouse/deckhouse/blob/35e1cfffcf6dc8a8599cc2995a2cdc412e717b5d/go_lib/set/set.go#L96-L103

Reply:
yes, thanks !